crates/precompile/src/bls12_381/g2_msm.rs

```diff
@@ -59,13 +65,26 @@ fn g2_msm(input: &Bytes, gas_limit: u64) -> PrecompileResult {
         );
     }
 
-    let points = p2_affines::from(&g2_points);
-    let multiexp = points.mult(&scalars, NBITS);
+    let out = if !g2_points.is_empty() {
+        let points = p2_affines::from(&g2_points);
+        let multiexp = points.mult(&scalars, NBITS);
 
-    let mut multiexp_aff = blst_p2_affine::default();
-    // SAFETY: multiexp_aff and multiexp are blst values.
-    unsafe { blst_p2_to_affine(&mut multiexp_aff, &multiexp) };
+        let mut multiexp_aff = blst_p2_affine::default();
+        // SAFETY: multiexp_aff and multiexp are blst values.
+        unsafe { blst_p2_to_affine(&mut multiexp_aff, &multiexp) };
 
-    let out = encode_g2_point(&multiexp_aff);
+        encode_g2_point(&multiexp_aff)
+    } else {
+        // default return size for empty msm
+        vec![0; 256].into()
+    };
     Ok((required_gas, out))
 }
+
+fn is_zero(buf: &[u8]) -> bool {
+    let (prefix, aligned, suffix) = unsafe { buf.align_to::<u128>() };
+
+    prefix.iter().all(|&x| x == 0)
+        && suffix.iter().all(|&x| x == 0)
+        && aligned.iter().all(|&x| x == 0)
```

Comment:
can't this just be one iter.all == 0 ?

Reply:
I have made this change, we can do it. Checking the infinity point is just checking if the slice is zero.